File: library/model/src/main/java/org/cru/godtools/model/jsonapi/ToolTypeConverter.java
package org.cru.godtools.model.jsonapi;

import org.ccci.gto.android.common.jsonapi.converter.TypeConverter;
import org.cru.godtools.model.Tool.Type;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public final class ToolTypeConverter implements TypeConverter<Type> {
    @Override
    public boolean supports(@NonNull final Class<?> clazz) {
        return Type.class.equals(clazz);
    }

    @Nullable
    @Override
    public String toString(@Nullable final Type value) {
        return value != null ? value.toJson() : null;
    }

    @Nullable
    @Override
    public Type fromString(@Nullable final String value) {
        return Type.fromJson(value);
    }
}

File: library/model/src/main/java/org/cru/godtools/model/jsonapi/ToolTypeConverter.kt
package org.cru.godtools.model.jsonapi

import org.ccci.gto.android.common.jsonapi.converter.TypeConverter
import org.cru.godtools.model.Tool

object ToolTypeConverter : TypeConverter<Tool.Type> {
    override fun supports(clazz: Class<*>) = Tool.Type::class.java == clazz
    override fun toString(value: Tool.Type?) = value?.toJson()
    override fun fromString(value: String?) = Tool.Type.fromJson(value)
}
